src/common.rs

```diff
@@ -188,3 +190,83 @@ pub struct Config {
     pub llvm_cxxflags: String,
     pub nodejs: Option<String>,
 }
+
+impl Default for Config {
+    fn default() -> Config {
+        let platform = rustc::session::config::host_triple().to_string();
+
+        Config {
+            compile_lib_path: PathBuf::from(""),
+            run_lib_path: PathBuf::from(""),
+            rustc_path: PathBuf::from("rustc"),
+            rustdoc_path: PathBuf::from("rustdoc-path"),
+            lldb_python: "python".to_owned(),
+            docck_python: "docck-python".to_owned(),
+            valgrind_path: None,
+            force_valgrind: false,
+            llvm_filecheck: None,
+            src_base: PathBuf::from("tests/run-pass"),
+            build_base: env::temp_dir(),
+            stage_id: "stage-id".to_owned(),
+            mode: Mode::RunPass,
+            run_ignored: false,
+            filter: None,
+            filter_exact: false,
+            logfile: None,
+            runtool: None,
+            host_rustcflags: None,
+            target_rustcflags: None,
+            target: platform.clone(),
+            host: platform.clone(),
+            gdb_version: None,
+            lldb_version: None,
+            llvm_version: None,
+            android_cross_path: PathBuf::from("android-cross-path"),
+            adb_path: "adb-path".to_owned(),
+            adb_test_dir: "adb-test-dir/target".to_owned(),
+            adb_device_status: false,
+            lldb_python_dir: None,
+            verbose: false,
+            quiet: false,
+            cc: "cc".to_string(),
+            cxx: "cxx".to_string(),
+            cflags: "cflags".to_string(),
+            llvm_components: "llvm-components".to_string(),
+            llvm_cxxflags: "llvm-cxxflags".to_string(),
+            nodejs: None,
+        }
+    }
+}
+
+/// Builder pattern for the `Config` struct
+#[derive(Default)]
+pub struct ConfigBuilder {
+    target_rustcflags: Option<String>,
+}
+
+impl ConfigBuilder {
```

Comment:
In principle we should add a method for each configurable parameter in `Config` right? Like

```
impl ConfigBuilder {
    fn compile_lib_path(mut self, p: PathBuf) -> ConfigBuilder {
        self.compile_lib_path = p;
        self
    }
    fn run_lib_path(mut self, p: PathBuf) -> ConfigBuilder { ... }
    // etc...
}
```